src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator<List<String>> {
+    @Override
+    public List<String> getReport() {
+        Map<String, Integer> fruitStorage = Storage.fruitStorage;
+        return fruitStorage.entrySet()
+                .stream()
+                .map(entry -> entry.getKey() + "," + entry.getValue())
+                .collect(Collectors.toList());
+    }
```

Comment:
The method `getReport` is not following the encapsulation principle. The `Storage.fruitStorage` should be private and accessed through public methods to ensure encapsulation is maintained.